src/server/api/controllers/products.controller.js

```diff
@@ -78,19 +57,48 @@ const getProductById = async (id) => {
         'is_on_discount',
         'discount_percent',
         'picture',
-      )
-      .where({ id });
-    if (products.length === 0) {
-      throw new Error(`incorrect entry with the id of ${id}`, 404);
+    )
+    .where('products.is_on_discount', '=', '1');
+
+const getProductById = async(prod_id) => {
+    const id = parseInt(prod_id, 10);
+    if (isNaN(prod_id) || id < 1) {
```

Comment:
Bonus: 
One thing we've covered in other PRs is that `parseInt` won't catch things like "1asd", it will simply return 1.

It's a bit hard to decide what to do in that case, 1 is a valid ID but the user submitted an invalid value.

To fully cover it, we'd have to add something like this above the current code:
```js
const NUMBER_REGEXP = /^\d+$/g; // Match any string that starts (^) and ends ($) with a number(\d), there can be one or more numbers
if (typeof product_id === 'string' && !product_id.match(NUMBER_REGEXP)) {
  // Throw error, it's a string but it's not a clean number, parseInt won't be safe
}
```